Bug: [Chassis] Expected Messages are missing from the syslog

### Is it platform specific

generic

### Importance or Severity

Critical

### Description of the bug

Expected Messages are missing from the syslog.

E               Expected Messages that are missing:
E               .*Caught exception while initializing thermal manager.*> 

E               Expected Messages that are missing:
E               .*Created ACL table.*


```
Bad case: TCP port 2514 is missing
=====================
admin@ixre-egl-board211:~$  netstat -anp | grep 514
(Not all processes could be identified, non-owned process info
will not be shown, you would have to be root to see it all.)
tcp        0      0 240.127.1.1:35144       240.127.1.2:6379        ESTABLISHED -                   
tcp        0      0 240.127.1.1:35146       240.127.1.2:6379        ESTABLISHED -                   
tcp        0      0 240.127.1.1:35144       240.127.1.3:6379        TIME_WAIT   -                   
tcp        0      0 240.127.1.1:45140       240.127.1.3:6379        ESTABLISHED -                   
udp        0      0 240.127.1.1:514         0.0.0.0:*                           -                   
unix  6      [ ]         DGRAM      CONNECTED     112514   -                    /dev/log
unix  3      [ ]         STREAM     CONNECTED     8514     -                    
unix  3      [ ]         STREAM     CONNECTED     51499    -                    /run/systemd/journal/stdout


Good Case: 
==========
admin@ixre-egl-board211:~$ netstat -anp | grep 514
tcp        0      0 240.127.1.1:2514        0.0.0.0:*               LISTEN      -                   
tcp        0      0 240.127.1.1:2514        240.127.1.1:37894       ESTABLISHED -                   
tcp       45      0 240.127.1.1:43018       240.127.1.1:2514        ESTABLISHED -                   
tcp        0      0 240.127.1.1:2514        240.127.1.2:37000       ESTABLISHED -                   
tcp        0      0 240.127.1.1:2514        240.127.1.3:37592       ESTABLISHED -                   
tcp        0      0 240.127.1.1:2514        240.127.1.3:37612       ESTABLISHED -                   
tcp        0      0 240.127.1.1:2514        240.127.1.1:43024       ESTABLISHED -                   
tcp        0      0 240.127.1.1:2514        240.127.1.2:37016       ESTABLISHED -                   
tcp        0      0 240.127.1.1:2514        240.127.1.2:36980       ESTABLISHED -                   
tcp        0      0 240.127.1.1:2514        240.127.1.3:37622       ESTABLISHED -           

```

### Steps to Reproduce

Load a new image.  All expected messages are int the syslog.
Reboot the dut a couple of times.  expected message are missing in the syslog.

### Actual Behavior and Expected Behavior

Actual: Expected Messages are missing from the syslog
Expected: Expected Messages are in the syslog


### Relevant log output

```shell

```

### Output of `show version`, `show techsupport`

```shell
Latest master image
```

### Attach files (if any)

_No response_

--- a/files/image_config/rsyslog/rsyslog-config.sh
+++ b/files/image_config/rsyslog/rsyslog-config.sh
@@ -68,6 +68,12 @@ if [ ! -f /etc/rsyslog.conf ] || ! cmp -s "$TMPFILE" /etc/rsyslog.conf; then
         exit 1
     fi
 else
-    # Config unchanged — just signal rsyslog to re-open log files
-    systemctl kill -s HUP rsyslog
+    if [[ ($NUM_ASIC -gt 1) ]]; then
+        # multi-asic, docker0 IP interface may not be present when rsyslog.service was started.
+        # restart the rsyslog for TCP port socket binding.
+        systemctl restart rsyslog
+    else
+        # Config unchanged — just signal rsyslog to re-open log files
+        systemctl kill -s HUP rsyslog
+    fi
 fi

